lms/views/api/sync.py

```diff
@@ -0,0 +1,43 @@
+import hashlib
+
+from pyramid.view import view_config
+
+from lms.models import HGroup, h_group_name
+
+
+@view_config(
+    route_name="api.sync", request_method="POST", renderer="json", permission="sync_api"
+)
+def sync(request):
+    authority = request.registry.settings["h_authority"]
+    canvas_api_svc = request.find_service(name="canvas_api_client")
+    lti_h_svc = request.find_service(name="lti_h")
+
+    context_id = request.json["course"]["context_id"]
+    custom_canvas_course_id = request.json["course"]["custom_canvas_course_id"]
+    tool_consumer_instance_guid = request.json["lms"]["tool_consumer_instance_guid"]
+
+    if request.lti_user.is_learner:
+        # For learners we only want the client to show the sections that the
+        # student belongs to, so fetch only the user's sections.
+        sections = canvas_api_svc.authenticated_users_sections(custom_canvas_course_id)
+    else:
+        # For non-learners (e.g. instructors, teaching assistants) we want the
+        # client to show all of the course's sections.
+        sections = canvas_api_svc.course_sections(custom_canvas_course_id)
```

Comment:
SpeedGrader will be a third case here, but we're not supporting that yet

Reply:
Very clear code here. The combination of properties and well encapsulated and named functions makes this very easy to understand.